src/aks-preview/azext_aks_preview/_helpers.py

```diff
@@ -21,11 +21,15 @@ def _populate_api_server_access_profile(api_server_authorized_ip_ranges, instanc
     else:
         profile = instance.api_server_access_profile
 
-    if api_server_authorized_ip_ranges == "":
+    if api_server_authorized_ip_ranges is None or api_server_authorized_ip_ranges == "":
         authorized_ip_ranges = []
     else:
         authorized_ip_ranges = [ip.strip() for ip in api_server_authorized_ip_ranges.split(",")]
 
+    if profile.enable_private_cluster and authorized_ip_ranges:
+        raise CLIError(
```

Comment:
Better to use ArgumentUsageError than CLIError.

Reply:
Changed in commit [#b8f0907](https://github.com/Azure/azure-cli-extensions/pull/4047/commits/b8f09070a8b2aa3b0aca338ae4f31c02b36418da).